src/utils/typedData.ts

```diff
@@ -305,15 +312,42 @@ export function encodeValue(
       } // else fall through to default
       return [type, getHex(data as string)];
     }
+    case 'i128': {
+      if (revision === Revision.Active) {
+        const value = BigInt(data as string);
+        assertRange(value, type, RANGE_I128);
+        return [type, getHex(value < 0n ? PRIME + value : value)];
+      } // else fall through to default
+      return [type, getHex(data as string)];
+    }
+    case 'timestamp':
+    case 'u128': {
+      if (revision === Revision.Active) {
+        assertRange(data, type, RANGE_U128);
+      } // else fall through to default
+      return [type, getHex(data as string)];
+    }
     case 'felt':
-    case 'bool':
-    case 'u128':
-    case 'i128':
-    case 'ContractAddress':
+    case 'shortstring': {
+      // TODO: should 'shortstring' diverge into directly using encodeShortString()?
```

Comment:
I want to draw attention to the approach taken with the `shortstring` encoding. It was implemented by reusing the existing `felt` encoding. 

As a basic example, a provided value `"hello"` will be encoded as `0x68656c6c6f`, this I believe to not be contentious. The part where I believe there might be issues with is the encoding of numerical strings, the existing `felt` encoding treats both numbers and numerical strings as the same value and such behaviour might be undesirable/unexpected for the `shortstring` type. As an example `"2"` will be encoded as `0x2` while someone coming from a Cairo background will probably assume it to be encoded as `0x32`.